pandas/tests/indexes/datetimes/test_ops.py

```diff
@@ -8,17 +8,88 @@
 from itertools import product
 import pandas as pd
 import pandas._libs.tslib as tslib
+from pandas._libs import period as libperiod
 import pandas.util.testing as tm
 from pandas import (DatetimeIndex, PeriodIndex, Series, Timestamp,
                     date_range, _np_version_under1p10, Index,
                     bdate_range)
 from pandas.tseries.offsets import BMonthEnd, CDay, BDay
+from pandas.tseries.frequencies import (RESO_DAY, RESO_HR, RESO_MIN, RESO_US,
+                                        RESO_MS, RESO_SEC)
 from pandas.tests.test_base import Ops
 
 
 START, END = datetime(2009, 1, 1), datetime(2010, 1, 1)
 
 
+class TestDatetimeIndexVectorizedTimestamp(object):
+    def test_timestamp_date_out_of_range(self):
+        # see gh-1475
+        pytest.raises(ValueError, DatetimeIndex, ['1400-01-01'])
+        pytest.raises(ValueError, DatetimeIndex, [datetime(1400, 1, 1)])
+
+    def test_timestamp_fields(self):
+        # extra fields from DatetimeIndex like quarter and week
+        idx = tm.makeDateIndex(100)
+
+        fields = ['dayofweek', 'dayofyear', 'week', 'weekofyear', 'quarter',
+                  'days_in_month', 'is_month_start', 'is_month_end',
+                  'is_quarter_start', 'is_quarter_end', 'is_year_start',
+                  'is_year_end', 'weekday_name']
+        for f in fields:
+            expected = getattr(idx, f)[-1]
+            result = getattr(Timestamp(idx[-1]), f)
+            assert result == expected
+
+        assert idx.freq == Timestamp(idx[-1], idx.freq).freq
+        assert idx.freqstr == Timestamp(idx[-1], idx.freq).freqstr
+
+    def test_round(self):
```

Comment:
move with similar tests

Reply:
The other test_round in this module already needs refactoring to use pytest.parametrize.  Will follow up.

Reply:
that’s fine but move it next to it